File: frontend/src/services/accessControl.ts
import { apiFetch } from "./api";
import type { AccessControl } from "../types/accessControl";

export async function getAccessControls(): Promise<AccessControl[]> {
  return apiFetch<AccessControl[]>("/access-control");
}

export async function getPrivilegedAccessControls(): Promise<
  AccessControl[]
> {
  return apiFetch<AccessControl[]>("/access-control/privileged");
}

export async function getHighRiskAccessControls(): Promise<
  AccessControl[]
> {
  return apiFetch<AccessControl[]>("/access-control/high-risk");
}

export async function getExcessiveAccessControls(): Promise<
  AccessControl[]
> {
  return apiFetch<AccessControl[]>("/access-control/excessive");
}

File: backend/app/api/access_control.py
from typing import List

from fastapi import APIRouter

from app.schemas.access_control import AccessControlResponse

from app.services.access_control_service import (
    get_all_access_controls,
    get_privileged_access_controls,
    get_high_risk_access_controls,
    get_excessive_access_controls,
)


router = APIRouter()


@router.get(
    "/access-control",
    response_model=List[AccessControlResponse],
)
def get_access_control():
    """
    Return access-control intelligence for all identities.
    """

    return get_all_access_controls()


@router.get(
    "/access-control/privileged",
    response_model=List[AccessControlResponse],
)
def get_privileged_access_control():
    """
    Return privileged access records.
    """

    return get_privileged_access_controls()


@router.get(
    "/access-control/high-risk",
    response_model=List[AccessControlResponse],
)
def get_high_risk_access_control():
    """
    Return high-risk access records.
    """

    return get_high_risk_access_controls()


@router.get(
    "/access-control/excessive",
    response_model=List[AccessControlResponse],
)
def get_excessive_access_control():
    """
    Return identities with excessive access.
    """

    return get_excessive_access_controls()